### PR TITLE
Use pino-abstract-transport

### PR DESCRIPTION
**Pull request by mcollina**
This change makes this module compatible with the new transport system.

This is the first step for https://github.com/pinojs/pino/issues/1106.

It is a semver-major change and it will require a minor change in pino@6 and pino@7 to achieve compatibility. before we exported the `prettyFactory`, now we export the function to build a transport, which includes split2.

### CODE REVIEW

**Comment by mcollina** [comment]
@jsumners I know you preferred this to not become compatible with the new system, but the changes are minimal so I thought it was worth a shot.

**Comment by mcollina** [comment]
cc @leorossi

File: index.js [hunk 1]
```diff
@@ -165,6 +169,41 @@ function prettyFactory (options) {
...
+    })
+
+    pump(source, stream, destination)
+    return stream
+  }, { parse: 'lines' })
+}
+
+module.exports = build
+module.exports.prettyFactory = prettyFactory
+module.exports.default = build
```
Comment (jsumners) [inline index.js:209]: This is semver major.

**Comment by mcollina** [comment]
@jsumners updated the require order.

**Review by jsumners — APPROVED** [review]

**Pull request merged**